### PR TITLE
refactor: split container env and add unit test

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: Allen Sun <allensun.shl@alibaba-inc.com>

### Ⅰ. Describe what this PR did

This PR tried to refactor code of updateContainerEnv function:

* split the function from container.go to be decoupled in container_env.go;
* split some detailed implementation to be encapsulated in some functions, like getLocalEnv, like combineLocalAndInputEnv.

This pr also adds some unit tests.

### Ⅱ. Does this pull request fix one issue?

none

### Ⅲ. Describe how you did it
none

### Ⅳ. Describe how to verify it
none

### Ⅴ. Special notes for reviews
@HusterWan

### CODE REVIEW

File: daemon/mgr/container_envs.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+	"github.com/alibaba/pouch/pkg/utils"
+
+	"github.com/magiconair/properties"
+	"github.com/sirupsen/logrus"
+)
+
+// container env is used to manage environment variables in container.
+// In original ways, processes in container can only accepted env when startup.
+// And container engine would store these data in container.json which is persistent on local disk.
```
Comment (HusterWan) [inline daemon/mgr/container_envs.go]: %s/these/those ?

File: daemon/mgr/container_envs.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+// Env used by rich container.
+func updateContainerEnv(inputRawEnv []string, baseFs string) error {
+	var (
+		envPropertiesPath = path.Join(baseFs, containerInstanceInfo)
+		envShPath         = path.Join(baseFs, pouchEnvFile)
+	)
+
+	// check the existence of related files.
+	// if dir of pouch.sh is not exist, it's unnecessary to update that files.
+	if _, err := os.Stat(path.Join(baseFs, pouchEnvDir)); err != nil {
```
Comment (HusterWan) [inline daemon/mgr/container_envs.go]: i think we can delete this part code. WDYT?

File: daemon/mgr/container_envs.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+	p, err := properties.LoadFile(envPropertiesPath, properties.ISO_8859_1)
+	if err != nil {
+		return fmt.Errorf("failed to properties load container's environment variable file(/etc/instanceInfo): %v", err)
+	}
+
+	for key, val := range newEnv {
+		if v, ok := p.Get("env_" + key); ok {
+			if key == "PATH" {
+				// refer to https://aone.alipay.com/project/532482/task/9745028
```
Comment (HusterWan) [inline daemon/mgr/container_envs.go]: it's my fault to add this url to PouchContainer's code, can you help me delete this line, thanks ? @allencloud

File: daemon/mgr/container_envs.go (hunk 2 above)
Comment (allencloud) [reply daemon/mgr/container_envs.go]: I do not think so. Since check dir, return nil if err; check file, return err if err.

**Comment by allencloud** [comment]
Unfortunately, my code change on the container env part always leads to cri-tools failure:
```
• Failure [0.816 seconds]
[k8s.io] Security Context
/home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/framework/framework.go:72
  SeccompProfilePath
  /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/security_context.go:461
    should support seccomp default which is unconfined on the container [It]
    /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/security_context.go:538
    Expected
        <string>: 
    to contain substring
        <string>: 0
    /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/security_context.go:1025
```

I do not think this is a flaky test with this pr, since I have rerun the cri-tools for several times and everytime, it fails to pass cri-tools.

Could you help to take a look at what happens. @Starnop @YaoZengzeng 
Thanks a lot.

**Comment by YaoZengzeng** [comment]
@allencloud I rerun it, it's OK right now.

The code in this PR seems to have nothing to do with CRI part 🤔

**Pull request merged**